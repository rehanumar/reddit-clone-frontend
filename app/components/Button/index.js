/**
 *
 * Button
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Button = styled.button`
  padding: ${props => props.padding || '4px 16px'};
  min-width: ${props => props.width || '32px'};
  min-height: ${props => props.height || '32px'};
  line-height: ${props => props.lineHeight || '17px'};
  font-weight: ${props => props.fontWeight || '700'};
  font-size: ${props => props.fontSize || '14px'};
  font-family: Noto Sans, Arial, sans-serif;
  background: ${props =>
    props.background ? props.theme[props.background] : props.theme.button};
  color: ${props =>
    props.color ? props.theme[props.color] : props.theme.body};
  fill: ${props => (props.fill ? props.theme[props.fill] : props.theme.body)};
  border: 1px solid
    ${props =>
    props.borderColor ? props.theme[props.borderColor] : props.theme.button};
  border-radius: 50px;
`;

const OrangeButton = styled(Button)`
  background: ${props => props.theme.button};
  border: 1px solid ${props => props.theme.button};
`;

const StyledLink = styled(Button)`
  text-decoration: none;
  color: ${props =>
    props.color ? props.theme[props.color] : props.theme.body};

  &:hover,
  &:visited,
  &:focus {
    text-decoration: none;
    color: ${props =>
    props.color ? props.theme[props.color] : props.theme.body};
  }
`;

function BtnLink(props) {
  const { onClick, children, ...rest } = props;
  const handleClick = e => {
    e.preventDefault();
    onClick();
  };
  return (
    <StyledLink as="a" role="button" href="" onClick={handleClick} {...rest}>
      {children}
    </StyledLink>
  );
}

BtnLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.object.isRequired,
};

export default Button;
export { OrangeButton, BtnLink };
