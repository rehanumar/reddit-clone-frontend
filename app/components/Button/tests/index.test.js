/**
 *
 * Tests for Button
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Button from '../index';

describe('<Button />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Button />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render with parent props', () => {
    const props = {
      background: 'button',
      color: 'body',
      fill: 'body',
      borderColor: 'button',
      lineHeight: '16',
      fontWeight: '12',
      width: '40px',
      height: '30px',
      padding: '20px 30px',
    };
    const {
      container: { firstChild },
    } = render(<Button {...props} />);
    expect(firstChild).toMatchSnapshot();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Button />);
    expect(firstChild).toMatchSnapshot();
  });
});
