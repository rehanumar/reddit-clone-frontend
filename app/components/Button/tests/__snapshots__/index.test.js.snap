// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with parent props 1`] = `
<button
  class="Button-boo9v0-0 gKLEuk"
  color="body"
  fill="body"
  font-weight="12"
  height="30px"
  width="40px"
/>
`;
